private/bufpkg/bufcheck/buflint/internal/buflintvalidate/numeric.go

```diff
@@ -0,0 +1,274 @@
+		fieldCount++
+		var convertErrorMessage string
+		fieldNumber := int32(field.Number())
+		switch fieldName := string(field.Name()); fieldName {
+		case "const":
+			constFieldNumber = fieldNumber
+			constant, convertErrorMessage, err = convertFunc(value)
+		case "gt":
+			gt, convertErrorMessage, err = convertFunc(value)
+			lowerBound = gt
+			lowerBoundFieldNumber = fieldNumber
+		case "gte":
+			gte, convertErrorMessage, err = convertFunc(value)
+			lowerBound = gte
+			lowerBoundFieldNumber = fieldNumber
+		case "lt":
+			lt, convertErrorMessage, err = convertFunc(value)
+			upperBound = lt
+			upperBoundFieldNumber = fieldNumber
+		case "lte":
+			lte, convertErrorMessage, err = convertFunc(value)
+			upperBound = lte
+			upperBoundFieldNumber = fieldNumber
+		case "in":
+			inFieldNumber = fieldNumber
+			for i := 0; i < value.List().Len(); i++ {
+				var converted *T
+				converted, convertErrorMessage, err = convertFunc(value.List().Get(i))
+				if converted != nil {
+					in = append(in, converted)
+				}
+			}
+		case "not_in":
+			for i := 0; i < value.List().Len(); i++ {
+				var converted *T
+				converted, convertErrorMessage, err = convertFunc(value.List().Get(i))
+				if converted != nil {
+					notIn = append(notIn, converted)
+				}
+			}
+		}
+		if convertErrorMessage != "" {
+			adder.addForPathf(
+				[]int32{ruleFieldNumber, fieldNumber},
+				"Field %q has an invalid %s: %s.",
+				adder.fieldName(),
+				adder.getFieldRuleName(ruleFieldNumber, int32(field.Number())),
+				convertErrorMessage,
+			)
+		}
+		return true
+	})
+	if err != nil {
+		return err
+	}
+	if constant != nil && fieldCount > 1 {
+		adder.addForPathf(
+			[]int32{ruleFieldNumber, constFieldNumber},
+			"Field %q has %s and does not need other rules in %s.",
+			adder.fieldName(),
+			adder.getFieldRuleName(ruleFieldNumber, constFieldNumber),
+			adder.getFieldRuleName(ruleFieldNumber),
+		)
+	}
+	if len(in) > 0 && fieldCount > 1 {
+		adder.addForPathf(
+			[]int32{ruleFieldNumber, inFieldNumber},
+			"Field %q has %s and does not need other rules in %s.",
+			adder.fieldName(),
+			adder.getFieldRuleName(ruleFieldNumber, inFieldNumber),
+			adder.getFieldRuleName(ruleFieldNumber),
+		)
+	}
+	if lowerBound == nil || upperBound == nil {
+		return nil
+	}
+	if gte != nil && lte != nil && compareFunc(upperBound, lowerBound) == 0 {
+		adder.addForPathsf(
+			[][]int32{
+				{ruleFieldNumber, lowerBoundFieldNumber},
+				{ruleFieldNumber, upperBoundFieldNumber},
+			},
+			"Field %q has equal %s and lte, use const instead.",
+			adder.fieldName(),
+			adder.getFieldRuleName(ruleFieldNumber, lowerBoundFieldNumber),
+		)
+		return nil
+	}
+	if compareFunc(upperBound, lowerBound) <= 0 {
+		adder.addForPathsf(
+			[][]int32{
+				{ruleFieldNumber, lowerBoundFieldNumber},
+				{ruleFieldNumber, upperBoundFieldNumber},
+			},
+			"Field %q has a %s (%v) higher than its %s (%v).",
```

Comment:
Same thing, example

Reply:
`Field "foo" has a (buf.validate.field).sfixed32.gte (11) higher than its (buf.validate.field).sfixed32.lte (10).`

Reply:
`Field %q has value %v for %s, which is higher than value %v for %s.`

Adjust other messages similarly.

Reply:
Now worded `must be lower`